main.rb

```diff
@@ -0,0 +1,144 @@
+# rubocop: disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/ModuleLength
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    i = 0
+    arr = to_a
+    while i < arr.length
+      yield(arr[i])
+      i += 1
+    end
+    self
+  end
+  [1, 2, 3].my_each { |n| puts "#{n}!" }
+
+  def my_each_each_with_index
+    return to_enum unless block_given?
+
+    i = 0
+    arr = to_a
+    while i < arr.length
+      yield(arr[i], i)
+      i += 1
+    end
+    self
+  end
+  [1, 2, 3].my_each_each_with_index { |_n, x| print x }
+
+  def my_select
+    return to_enum unless block_given?
+
+    arr = to_a
+    final_array = []
+    arr.my_each { |x| final_array.push(x) if yield(x) }
+    final_array
+  end
+  [1, 2, 3, 4, 5, 6, 7, 8, 9, 10].my_select { |x| puts x if x.even? }
+
+  def my_all?(arg = nil)
+    arr = to_a
+    if block_given?
+      arr.my_each { |x| return true unless yield(x) == false }
+      return true
+    elsif arg.nil?
+      return false
+    elsif !arg.nil? && (arg.is_a? Regexp)
+      arr.my_each { |x| return false unless x.match(arg) }
+    elsif (arg.is_a? Class) && !arg.nil?
+      arr.my_each { |x| return false unless [x.class, x.class.superclass].include?(arg) }
+    else
+      arr.my_each { |i| return false if i != arg }
+
+    end
+    true
+  end
+
+  def my_any?(arg = nil)
+    if block_given?
+      my_each do |x|
+        return true if yield x
+      end
+
+    elsif !arg.nil? && (arg.is_a? Class)
+      my_each do |x|
+        return false unless x.class == arg
+      end
+
+    elsif (arg.class == Regexp) && !arg.nil?
+      my_each do |x|
+        return false unless arg.match(x)
+      end
+
+    else
+      my_each do |x|
+        return true if x == arg
+      end
+
+    end
+    false
+  end
+
+  def my_none?
+    return to_enum unless block_given?
+
+    !my_any?
+  end
```

Comment:
Your my_none seems unable to handle arguments at it's input. Also, the errors from my_any are propagating to my_none. Please fix `my_any` in order to fix `my_none`:

```
  7) enumerables #my_none? returns false if the block ever returns true for all elements
     Failure/Error: expect(array.my_none?(&false_block)).to eq(array.none?(&false_block))
     
       expected: false
            got: true
     
       (compared using ==)
     
       Diff:
       @@ -1 +1 @@
       -false
       +true
       
     # ./spec/enumerables_spec.rb:282:in `block (3 levels) in <top (required)>'

  8) enumerables #my_none? returns false if the block ever returns true for all elements::range
     Failure/Error: expect(range.my_none?(&false_block)).to eq(range.none?(&false_block))
     
       expected: false
            got: true
     
       (compared using ==)
     
       Diff:
       @@ -1 +1 @@
       -false
       +true
       
     # ./spec/enumerables_spec.rb:286:in `block (3 levels) in <top (required)>'

  9) enumerables #my_none? when no block or argument is given returns true only if none of the collection members is true
     Failure/Error: expect(false_array.my_none?).to be true
     
       expected true
            got #<Enumerator:47356367743860> => #<Enumerator: [nil, false, nil, false]:each>
     # ./spec/enumerables_spec.rb:296:in `block (4 levels) in <top (required)>'

  10) enumerables #my_none? when no block or argument is given returns false only if one of the collection members is true
      Failure/Error: expect(true_array.my_none?).to be false
      
        expected false
             got #<Enumerator:47356367769740> => #<Enumerator: [nil, false, true, []]:each>
      # ./spec/enumerables_spec.rb:300:in `block (4 levels) in <top (required)>'

  11) enumerables #my_none? when a class is passed as an argument returns true if none of the collection is a member of such class
      Failure/Error: expect(array.my_none?(String)).to be array.my_none?(String)
      
      ArgumentError:
        wrong number of arguments (given 1, expected 0)
      # ./main.rb:83:in `my_none?'
      # ./spec/enumerables_spec.rb:306:in `block (4 levels) in <top (required)>'

  12) enumerables #my_none? when a class is passed as an argument returns true if none of the collection is a member of such class::Numeric
      Failure/Error: expect(array.my_none?(Numeric)).to be array.my_none?(Numeric)
      
      ArgumentError:
        wrong number of arguments (given 1, expected 0)
      # ./main.rb:83:in `my_none?'
      # ./spec/enumerables_spec.rb:310:in `block (4 levels) in <top (required)>'

  13) enumerables #my_none? when a class is passed as an argument returns false if any of the collection is a member of such class
      Failure/Error: expect(array.my_none?(String)).to be array.my_none?(String)
      
      ArgumentError:
        wrong number of arguments (given 1, expected 0)
      # ./main.rb:83:in `my_none?'
      # ./spec/enumerables_spec.rb:315:in `block (4 levels) in <top (required)>'

  14) enumerables #my_none? when a Regex is passed as an argument returns true if none of the collection matches the Regex
      Failure/Error: expect(words.my_none?(/z/)).to be words.none?(/z/)
      
      ArgumentError:
        wrong number of arguments (given 1, expected 0)
      # ./main.rb:83:in `my_none?'
      # ./spec/enumerables_spec.rb:321:in `block (4 levels) in <top (required)>'

  15) enumerables #my_none? when a Regex is passed as an argument returns false if any of the collection matches the Regex
      Failure/Error: expect(words.my_none?(/d/)).to be words.none?(/d/)
      
      ArgumentError:
        wrong number of arguments (given 1, expected 0)
      # ./main.rb:83:in `my_none?'
      # ./spec/enumerables_spec.rb:325:in `block (4 levels) in <top (required)>'

  16) enumerables #my_none? when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
      Failure/Error: expect(words.my_none?(5)).to be words.none?(5)
      
      ArgumentError:
        wrong number of arguments (given 1, expected 0)
      # ./main.rb:83:in `my_none?'
      # ./spec/enumerables_spec.rb:331:in `block (4 levels) in <top (required)>'

  17) enumerables #my_none? when a pattern other than Regex or a Class is given returns false only if one of the collection matches the pattern
      Failure/Error: expect(words.my_none?(5)).to be words.none?(5)
      
      ArgumentError:
        wrong number of arguments (given 1, expected 0)
      # ./main.rb:83:in `my_none?'
      # ./spec/enumerables_spec.rb:335:in `block (4 levels) in <top (required)>'

```